pkg/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/customprops.go

```diff
@@ -30,7 +30,7 @@ type CustomPropertiesModifier struct {
 }
 
 func (mod CustomPropertiesModifier) Enabled() bool {
-	return mod.hasCustomProperties()
+	return mod.hasCustomProperties() || mod.dk.NeedsCustomNoProxy()
```

Comment:
why not add the ` mod.dk.NeedsCustomNoProxy()` check inside the `hasCustomProperties` instead?

edit: you actually did that, then why do it again 😅